pkg/osbuild/tar_stage.go

```diff
@@ -70,6 +85,25 @@ func (o TarStageOptions) validate() error {
 		}
 	}
 
+	if o.Compression != "" {
+		allowedArchiveCompressionValues := []TarArchiveCompression{
+			TarArchiveCompressionAuto,
+			TarArchiveCompressionXz,
+			TarArchiveCompressionGzip,
+			TarArchiveCompressionZstd,
+		}
+		valid := false
+		for _, value := range allowedArchiveCompressionValues {
+			if o.Compression == value {
+				valid = true
+				break
+			}
+		}
+		if !valid {
+			return fmt.Errorf("'compression' option does not allow %q as a value", o.Compression)
```

Comment:
Wrapping errors without values does not allow the caller to use `errors.Is`, ideally there should be a new error value `ErrUnuspportedCompression` wrapped into this. But probably not big deal here since the caller will just error out.

Reply:
Personally, I do not see any value in wrapping the error at the moment just because it is possible. We never inspect the error type returned from stage options validation to take a specific action based on it.